pkg/ffmpeg/encoder_scene_preview_chunk.go

```diff
@@ -14,12 +14,43 @@ type ScenePreviewChunkOptions struct {
 	OutputPath string
 }
 
-func (e *Encoder) ScenePreviewVideoChunk(probeResult VideoFile, options ScenePreviewChunkOptions, preset string) {
+func (e *Encoder) ScenePreviewVideoChunk(probeResult VideoFile, options ScenePreviewChunkOptions, preset string, fallback bool) error {
+	fallbackMinSlowSeek := 20.0
+	fastSeek := options.StartTime
+	slowSeek := 0.0
+
 	args := []string{
 		"-v", "error",
-		"-xerror",
-		"-ss", strconv.FormatFloat(options.StartTime, 'f', 2, 64),
-		"-i", probeResult.Path,
+	}
+
+	// Not fallback: enable xerror
+	if !fallback {
+		args = append(args, "-xerror")
+	} else {
+		// Fallback try a combination of fast/slow seek instead of only fastseek
+		if fastSeek > fallbackMinSlowSeek {
+			fastSeek = fastSeek - fallbackMinSlowSeek
+			slowSeek = fallbackMinSlowSeek
+		} else {
+			slowSeek = fastSeek
+			fastSeek = 0
+		}
```

Comment:
I think this needs some further documentation. It's not clear to me what it being done here and why. Also need to rationale for the `fallbackMinSlowSeek` value. This seems like it would give some unexpected behaviour if the video file is particularly short.

Reply:
I've slightly refactored the code to make it easier to understand, and commented it better.